src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1841,7 +1776,11 @@ void TorrentHandle::handleSaveResumeDataAlert(const lt::save_resume_data_alert *
     resumeData["qBt-queuePosition"] = (static_cast<int>(nativeHandle().queue_position()) + 1); // qBt starts queue at 1
     resumeData["qBt-hasRootFolder"] = m_hasRootFolder;
 
-    if (m_pauseWhenReady) {
+#if (LIBTORRENT_VERSION_NUM < 10200)
+    if (m_nativeStatus.stop_when_ready) {
+#else
+    if (bool {m_nativeStatus.flags & lt::torrent_flags::stop_when_ready}) {
```

Comment:
IIRC the flags type has implicit conversion to `bool` so the "bool" word can be omitted if you want.

Reply:
>BTW it is quite common (in embedded systems) to omit the explicit cast when doing low-level bit operations and I some how believe libtorrent had this in mind when designing the flag type interface.

It just behaves as regular integral type.
This common pattern of using flags (`if (flags & some_flag)`) is valid only in one case (when you test for single flag). Someone may have a misunderstanding and generate an incorrect expression for testing multiple flags.

Reply:
Anyway I have addressed all your comments.